core/types/rollup_cost.go

```diff
@@ -99,6 +106,11 @@ type StateGetter interface {
 // sender of non-Deposit transactions.  It returns nil if no data availability fee is charged.
 type L1CostFunc func(rcd RollupCostData, blockTime uint64) *big.Int
 
+// OperatorCostFunc is used in the state transition to determine the operator fee charged to the
+// sender of non-Deposit transactions. It returns nil if no data availability fee is charged.
```

Comment:
Is there a particular reason to return `nil` if no fee is charged rather than 0? The actual implementation doesn't ever seem to actually return `nil` and we have inconsistent nil checks where it's used.

Reply:
Fixed